### PR TITLE
Path through array in MockSmartStore.js

### PR DESCRIPTION
**Pull request by wmathurin**
Also testing queries that go against a non-leaf node (array nodes are just one example)
When I run those tests from iOS/Android, changes might be required

### CODE REVIEW

File: test/MockSDKInfo.js [hunk 1]
```diff
@@ -47,7 +47,7 @@ var MockSDKInfo = (function(window) {
             cordova.interceptExec(SDKINFO_SERVICE, "getInfo", function (successCB, errorCB, args) {
                 successCB(new SDKInfo("4.1.0", 
                                       ["com.salesforce.oauth", "com.salesforce.sdkinfo", "com.salesforce.sfaccountmanager", "com.salesforce.testrunner", "com.salesforce.smartstore", "com.salesforce.smartsync"], 
-                                      "SalesforceHybridTest", "1.0",
```
Comment (wmathurin) [inline test/MockSDKInfo.js]: Unrelated to the rest of the PR, just noticed this test is not passing in a browser

File: test/MockSmartStore.js [hunk 2]
```diff
@@ -282,15 +282,46 @@ var MockSmartStore = (function(window) {
...
+				    }
+			    }
+			    else {
+				    result = this.projectHelper(jsonObj[pathElement], pathElements, index+1);
+			    }
+		    }
+
+		    return result;
+	    },
+
```
Comment (wmathurin) [inline test/MockSmartStore.js]: Same change we already made on iOS and Android

File: test/MockSDKInfo.js (hunk 1 above)
Comment (bhariharan) [inline test/MockSDKInfo.js]: Interesting! I thought I had fixed this a while ago. Guess not.

**Comment by bhariharan** [comment]
:+1:

**Pull request merged**